src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -55,75 +82,83 @@ const SMOOTHING_KERNEL: Float32Array = _getSmoothingKernel(
+ * fills by mutating the selected pecentage buffers and the category fill values.
  */
-export class ActivityGraphFills {
-  rangeStart: Milliseconds;
-  rangeEnd: Milliseconds;
-  rangeLength: Milliseconds;
-  canvasPixelWidth: DevicePixels;
-  canvasPixelHeight: DevicePixels;
-  devicePixelRatio: number;
-  xPixelsPerMs: number;
-  categoryDrawStyles: CategoryDrawStyle[];
-  percentageBuffers: SelectedPercentageAtPixelBuffers[];
-  samples: SamplesTable;
-  stackTable: StackTable;
-  interval: Milliseconds;
-  greyCategoryIndex: IndexIntoCategoryList;
-  samplesSelectedStates: ?Array<boolean>;
-  categoryFills: CategoryFill[];
-  treeOrderSampleComparator: ?(
-    IndexIntoSamplesTable,
-    IndexIntoSamplesTable
-  ) => number;
+export class ActivityGraphFillComputer {
+  +lastRenderedGraphSettings: LastRenderedGraphSettings;
+  // The fills and percentages are mutated in place.
+  +mutablePercentageBuffers: SelectedPercentageAtPixelBuffers[];
+  +mutableFills: CategoryFill[];
 
   constructor(
-    canvasPixelWidth: DevicePixels,
-    canvasPixelHeight: DevicePixels,
-    {
-      rangeEnd,
-      rangeStart,
-      categories,
-      interval,
-      samplesSelectedStates,
-      treeOrderSampleComparator,
-      fullThread: { samples, stackTable },
-    }: ActivityGraphProps,
-    categoryDrawStyles: CategoryDrawStyle[]
+    lastRenderedGraphSettings: LastRenderedGraphSettings,
+    mutablePercentageBuffers: SelectedPercentageAtPixelBuffers[],
+    mutableFills: CategoryFill[]
   ) {
-    // Collect the common variables used on the various methods.
-    this.canvasPixelWidth = canvasPixelWidth;
-    this.canvasPixelHeight = canvasPixelHeight;
-    this.rangeEnd = rangeEnd;
-    this.rangeStart = rangeStart;
-    this.rangeLength = rangeEnd - rangeStart;
-    this.categoryDrawStyles = categoryDrawStyles;
-    this.interval = interval;
-    this.xPixelsPerMs = this.canvasPixelWidth / this.rangeLength;
-    this.samples = samples;
-    this.stackTable = stackTable;
-    this.samplesSelectedStates = samplesSelectedStates;
-    this.greyCategoryIndex = categories.findIndex(c => c.color === 'grey') || 0;
-    this.devicePixelRatio = window.devicePixelRatio;
-    this.treeOrderSampleComparator = treeOrderSampleComparator;
-    this.percentageBuffers = _createSelectedPercentageAtPixelBuffers(
-      categoryDrawStyles,
-      this.canvasPixelWidth
-    );
-    this.categoryFills = _getCategoryFills(
-      categoryDrawStyles,
-      this.percentageBuffers
-    );
+    this.lastRenderedGraphSettings = lastRenderedGraphSettings;
+    this.mutablePercentageBuffers = mutablePercentageBuffers;
+    this.mutableFills = mutableFills;
+  }
+
+  /**
+   * Run the computation to compute a list of the fills that need to be drawn for the
+   * ThreadActivityGraph.
+   */
+  run(): void {
+    const {
+      mutableFills,
+      lastRenderedGraphSettings: { canvasPixelWidth },
+    } = this;
```

Comment:
I'd move this below the call to `this.accumulateSampleCategories()`. I find it a bit strange to read a member variable (`mutableFills`) just before it is mutated in a different member function (`accumulateSampleCategories`) and then use it afterwards.